src/Collection.js

```diff
@@ -1,37 +1,42 @@
 // @flow
-import { observable, action, computed, IObservableArray, runInAction, toJS } from 'mobx'
+import { observable, action, computed, IObservableArray } from 'mobx'
 import Model from './Model'
-import {
-  isEmpty,
-  filter,
-  isMatch,
-  find,
-  difference,
-  debounce,
-  map,
-  last
-} from 'lodash'
-import ErrorObject from './ErrorObject'
-import Request from './Request'
+import { filter, isMatch, find, difference, map } from 'lodash'
 import apiClient from './apiClient'
-import type { Label, CreateOptions, SetOptions, Id } from './types'
+import Base from './Base'
+import Request from './Request'
+import type { CreateOptions, SetOptions, GetOptions, FindOptions, Id } from './types'
 
-export default class Collection<T: Model> {
-  @observable request: ?Request = null
-  @observable error: ?ErrorObject = null
-  @observable models: IObservableArray<T> = []
+export default class Collection extends Base {
+  @observable models: IObservableArray<Model> = []
```

Comment:
Honestly, all I know about flow is what i've been learning to work on this library, so if you think it's a regression, it probably is. It would be great if you can help me with that, I really struggled with flow on this PR 😵

Reply:
I can work on that myself then.